sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1495,6 +1495,56 @@ pi_result piDevicesGet(pi_platform Platform, pi_device_type DeviceType,
   return PI_SUCCESS;
 }
 
+// sub-sub-device
+pi_result getComputeCmdQueueOrdinals(
+    pi_device PiSubDevice, std::vector<int> &Ordinals,
+    std::vector<ze_command_queue_group_properties_t> &AllQueueProperties) {
+  uint32_t numQueueGroups = 0;
+  ZE_CALL(zeDeviceGetCommandQueueGroupProperties,
+          (PiSubDevice->ZeDevice, &numQueueGroups, nullptr));
+  if (numQueueGroups == 0) {
+    return PI_ERROR_UNKNOWN;
+  }
+  std::vector<ze_command_queue_group_properties_t> QueueProperties(
+      numQueueGroups);
+  ZE_CALL(zeDeviceGetCommandQueueGroupProperties,
+          (PiSubDevice->ZeDevice, &numQueueGroups, QueueProperties.data()));
+
+  bool noComputeEngineFlag = true;
+  for (uint32_t i = 0; i < numQueueGroups; i++) {
+    if (QueueProperties[i].flags &
+        ZE_COMMAND_QUEUE_GROUP_PROPERTY_FLAG_COMPUTE) {
+      Ordinals.push_back(i);
+      AllQueueProperties.push_back(QueueProperties[i]);
+      noComputeEngineFlag = false;
+    }
+  }
+
+  if (noComputeEngineFlag) {
+    return PI_ERROR_UNKNOWN;
+  }
+
+  return PI_SUCCESS;
+}
+
+pi_result initializeWithOrdinal(
+    pi_device PiSubSubDevice, int Ordinal,
+    std::vector<ze_command_queue_group_properties_t> &QueueProperties) {
+  PiSubSubDevice->ZeComputeQueueGroupIndex = Ordinal;
+  PiSubSubDevice->ZeComputeQueueGroupProperties = QueueProperties[Ordinal];
```

Comment:
Could you confirm this logic is correct?. in line 1517, you are pushing the i to Ordinals, and i could be from 0 to numQueueGroups -1  However, AllQueueProperties has another range, from 0 to the number of queue groups with COMPUTE flag, so using i to select an element in the AllqueueProperties vector might not be correct.

Reply:
right, will be using a map for AllqueueProperties instead of a vector..

Reply:
done.